DataStructures/Timeline.cs

```diff
@@ -0,0 +1,554 @@
+/*
+    Author: Lorenzo Lotti
+    Name: Timeline (DataStructures.Timeline<TValue>)
+    Type: Data structure (class)
+    Description: A collection of dates/times and values sorted by dates/times easy to query.
+    Usage:
+        this data structure can be used to represent an ordered series of dates or times with which to associate values.
+        An example is a chronology of events:
+            306: Constantine is the new emperor,
+            312: Battle of the Milvian Bridge,
+            313: Edict of Milan,
+            330: Constantine move the capital to Constantinople.
+*/
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DataStructures
+{
+    /// <summary>
+    /// A collection of <see cref="DateTime"/> and <see cref="TValue"/> sorted by <see cref="DateTime"/> field.
+    /// </summary>
+    /// <typeparam name="TValue">Value associated with a <see cref="DateTime"/>.</typeparam>
+    public class Timeline<TValue>:
+        IEnumerable<(DateTime Time, TValue Value)>,
```

Comment:
Codacy found an issue: ['ICollection<(DateTime Time, TValue Value)>' implements 'IEnumerable<(DateTime Time, TValue Value)>' so 'IEnumerable<(DateTime Time, TValue Value)>' can be removed from the inheritance list.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5574777)